Remove slimmer/static integration from Feedback

KludgeKML opened the pull request:
https://trello.com/c/0o6io1aA/344-remove-slimmer-feedback, [Jira issue PNP-6416](https://gov-uk.atlassian.net/browse/PNP-6416)

Do not merge before:
- [x] https://github.com/alphagov/govuk-helm-charts/pull/3249

Example pages:
- https://feedback-pr-2108.herokuapp.com/contact
- https://feedback-pr-2108.herokuapp.com/done/gweld-atwrneiaeth-arhosol

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

KludgeKML committed:
Remove slimmer gem and config

- Note that the existing template for this is gem_layout_no_feedback_form, so
  our application layout will want to replicate that (it is the base gem layout with
  `omit_feedback_form: true`.)

Audit trail:
https://github.com/alphagov/static/blob/main/app/views/root/gem_layout_no_feedback_form.html.erb

KludgeKML committed:
Add basic elements bundle

- although we could bring in these elements as individual stylesheets, in
  email-alert-frontend we got considerable benefit by bundling everything
  that was used in all pages on the app, so we start off by doing that here.
- basic component stylesheets in application.scss
- anything in application.scss also needs to be added to the custom
  exclude list so that it isn't included twice by the components gem.
- finally, the JS files needed for those components (previously supplied
  by static)

KludgeKML committed:
Add basic layout with direct component calls

- Update layout to use layout_for_public component directly
- we don't need the #wrapper div, because that was for static/slimmer to
  replace, but we do need a govuk-width-container around the
  breadcrumbs.
- to optimise the contents of link headers, we create the stylesheet
  link tag at the top of the view.

KludgeKML committed:
Update content_for blocks for head content

- anything that was extra_head_content now needs to be content for head
- anything that was extra_headers would just not have been working (there
  wasn't a yield to them), so update to head and confirm that this is correct.

KludgeKML committed:
Add application.scss to dartsass config

andysellick commented on app/views/layouts/application.html.erb:
I don't think this should be a problem but raising just in case... some of the attributes on elements aren't wrapped in quotes in the new code, example below (might be just because it's running locally compared with live).

![Screenshot 2025-04-30 at 10 38 35](https://github.com/user-attachments/assets/b9028237-b104-4ba2-b37f-d3fe0c3413e3)

andysellick replied:
Fixed, see commit.

andysellick committed:
Fix missing quote marks around HTML attributes

- in the old world, slimmer inserted any missing ones automatically thanks to nokogiri parsing of the page content
- but that doesn't happen anymore, so manually inserting them where they should be

andysellick commented on app/views/layouts/service_feedback.html.erb:
```diff
@@ -1,44 +1,50 @@
+
+<%= content_for :head do %>
+  <%= csp_meta_tag %>
+  <meta name="robots" content="noindex">
```
Fixed, see commit.

andysellick commented on app/views/layouts/service_feedback.html.erb:
```diff
@@ -1,44 +1,50 @@
+
+<% I18n.locale = I18n.default_locale %>
+<%= render "govuk_publishing_components/components/layout_for_public", {
+      title: yield(:title),
```
Fixed, see commit.

andysellick committed:
Add rendering-app metatag

- This metatag was added by slimmer in the past, and appears to be necessary for
  the browser extension to know the correct rendering app.
- It clashes with the render_app meta tag, ideally we should standardise so only
  one is necessary, but for the moment this will do.

andysellick commented on app/views/layouts/service_feedback.html.erb:
```diff
@@ -1,44 +1,50 @@
+
+<%= content_for :head do %>
+  <%= csp_meta_tag %>
+  <meta name="robots" content="noindex">
```
Fixed, see commit.

andysellick commented on app/views/layouts/service_feedback.html.erb:
```diff
@@ -1,44 +1,50 @@
+
+<% content_for :body do %>
+  <div class="completed_transaction service govuk-width-container">
+    <%= render 'govuk_publishing_components/components/breadcrumbs', {
```
Fixed, see commit.

andysellick committed:
Add govuk_web_banners gem

- Add banner JS dependencies
- Add global/emergency banner support in layouts
- Add initializer for emergency banner REDIS support
- Mock redis client in all tests (we're not explicitly
  testing emergency banner support here, since we assume
  that's tested adequately in the gem).

Co-authored-by: KludgeKML <keith.lawrence@digital.cabinet-office.gov.uk>

KludgeKML committed:
feat: add middleware to optimise GA4 data attributes

KludgeKML committed:
Update service_feedback layout

- Note that for this layout the robots file is noindex, nofollow. Previously
  this was inserted by the shared completed_transaction_feedback
  partial, but that's only used in this view and everything that uses this
  view also uses that partial, so it can just be moved over.
